src/@itwinjs-sandbox/hooks/useSampleWidget.ts

```diff
@@ -36,6 +39,21 @@ export const useSampleWidget = (instructions: string, iModelList?: SampleIModels
     }
   }, [iModelList, sampleIModels]);
 
+  /**
+   * Fix to add Hilite ruleset to presentation until BUG #599922 is addressed
+   */
+  useEffect(() => {
+    const addHiliteRuleset = async () => {
+      try {
+        if (Presentation.presentation)
+          await Presentation.presentation.rulesets().add((HILITE_RULESET as any).default as Ruleset);
+      } catch {
+        // Presentation not initialized
+      }
+    };
+    addHiliteRuleset();
+  });
```

Comment:
I suggest adding [] as your dependencies on this useEffect, otherwise this will get called on every state change